### DISCUSSION TITLE
How to print Barcode in fixed dimension container

### DISCUSSION THREAD

**Discussion by niravbhimani53** [QuestPDF/QuestPDF#928] [Q&A] [answered]
Hello devs, I need some help.

I would like to print the barcode in a container of fixed dimensions where the barcode should take full width and height of the container. But I am unable to do it. I have tried increasing the font size, `AspectRatio` and `Scale`, but nothing worked out.

The barcode is always small in size and increasing font size throws an error because the container can't fit.

```c#
public void Compose(IDocumentContainer container)
        {
            FontManager.RegisterFont(File.OpenRead("fonts/LibreBarcode39-Regular.ttf"));
            container
                .Page(page =>
                {
                    page.Margin(10, Unit.Millimetre);
                    page.Size((float)200, (float)300, Unit.Millimetre);
                    page
                        .Content()
                        .Element(element =>
                        {
                            element
                                .Width(30, Unit.Millimetre)
                                .Height(5, Unit.Millimetre)
                                .Border(1)
                                .Text($"*12345678*")
                                .FontFamily("Libre Barcode 39")
                                .FontSize(10);
                        });
                });
        }
```
<img width="971" alt="snip" src="https://github.com/QuestPDF/QuestPDF/assets/12537912/97491f83-5581-470e-8bb1-c6f54097bcfe">

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The sizing trouble comes from using a barcode font rendered as text, where font size does not map cleanly to a fixed container and overflows. The recommended approach is QuestPDF's native barcode support via the ZXing.Net library, which renders the barcode as an SVG that fills the container you give it. You set the box with `.Width(...)` and `.Height(...)`, and the barcode scales to fit — no font-size adjustment needed. See the [Barcodes](https://www.questpdf.com/api-reference/barcodes.html) documentation for a full example.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
